pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -184,6 +191,50 @@ protected ConsumerBase(PulsarClientImpl client, String topic, ConsumerConfigurat
         return internalReceive(timeout, unit);
     }
 
+    @Override
+    public CompletableFuture<Message<T>> receiveAsync(int timeout, TimeUnit unit) {
+
+        if (listener != null) {
+            return FutureUtil.failedFuture(new PulsarClientException.InvalidConfigurationException(
+                    "Cannot use receive() when a listener has been set"));
+        }
+        try {
+            verifyConsumerState();
+        } catch (PulsarClientException e) {
+            return FutureUtil.failedFuture(e);
+        }
+
+        ScheduledExecutorService messageScheduledFuture = Executors.newSingleThreadScheduledExecutor(
```

Comment:
1. In this way, you break the former logic handling the case when queue size is not zero.
2. We just need remove the code below:
```
if (conf.getReceiverQueueSize() == 0) {
    throw new PulsarClientException.InvalidConfigurationException(
            "Can't use receive with timeout, if the queue size is 0");
}
```
and then override the method `internalReceive(timeout, unit)` in `ZeroQueueConsumerImpl`
3. use single instance of `ScheduledExecutorService`.

Reply:
@complone I left some comments, please take a look.
and then you can ping me again.

Reply:
Okay, thank you for your attention to this issue. There are some things that need to be dealt with recently, it will be done as soon as possible this week

Reply:
@linlinnn Hi 
I noticed that user @eolivelli mentioned that we should provide users with options for their own configuration. I think we should refer to the implementation of pulsar's other timing thread pools and let users set how long they need to wait when the elements of the receiving queue are 0.

originally posted by @eolivelli in https://github.com/apache/pulsar/issues/10479#issuecomment-831996071_

Relevant patches will be submitted in the next few days, thank you for your guidance

Reply:
> can you please clarify the description of this work ?
> it looks interesting but I do not fully understand the goal

@eolivelli In order to achieve the collection of a message within a given time as you mentioned, allowing users to use the Java client to use receiverQueueSize(0) to create consumers when receiving overtime